client/signup/config/flows.js

```diff
@@ -269,8 +269,23 @@ function filterFlowName( flowName ) {
 	return flowName;
 }
 
+function filterDestination( destination, dependencies ) {
+	if ( abtest( 'guidedTours' === 'guided' ) ) {
+		const tourName = 'main';
+
+		if ( dependenciesContainCartItem( dependencies ) ) {
+			return getCheckoutUrl( dependencies ) + `?tour=${ tourName }`;
+		}
+
+		return `/posts/${ dependencies.siteSlug }/?tour=${ tourName }`;
```

Comment:
Maybe we can check `isExternal( destination )` here, so for everything but external destinations, we'd land on the original Calypso destination?